### PR TITLE
New options for training: ConvNeXt framework, early stopping with patience, and extra augmentations

### PR DESCRIPTION
**Pull request by PetervanLunteren**
Added extra options that can be user specified during training:

1. **ConvNeXt** - I added the 'convnext_base' model with pre-trained ImageNet weights to retrain from. This model architecture has proven very efficient (see [Deepfaune paper](https://link.springer.com/article/10.1007/s10344-023-01742-7)). Can be chosen by specifying `architecture: "convnext_base"` in config file.
2. **Early stopping** - There is now the option to set a patience value. If specified (e.g., `patience: 10`) it will save `last.pt` each epoch and `best.pt` only if it has a lower val loss than the previous best model. 
3. **Augmentations** - I added the option to add extra augmentations if desired. These augmentation methods are also used in the Deepfaune model (see [Deepfaune paper](https://link.springer.com/article/10.1007/s10344-023-01742-7)). Can be specified with `augment: True` in the config file. Will only affect the train batches, never test or val. 

I hope these suggestions prove to be valuable!

### CODE REVIEW

**Comment by PetervanLunteren** [comment]
I've added a bunch more like EfficientNetV2 small, auto batch size calculations, class-specific test reports, schedulers, class weights, etc. Let me know if you would like me to PR them.

**Comment by iingram** [comment]
Hi Peter! Thanks for this PR! I just thought I would chime in to let you know that Kyra, who I think you know and who is the main developer on this project, has been on holiday and will be for a couple more weeks. So, it's not that we aren't super grateful to receive contributions, and eager to potentially merge them, it is just the season of people being on holidays! :-) Maybe Kyra has told you as much on some other channel.

**Comment by PetervanLunteren** [comment]
Hi @iingram, thanks for your message! No problem, just let me know in due time. Happy to contribute :)

**Review by tkswanson — APPROVED** [review]
Great edits, thanks for helping flesh out the training protocols!  I saw the comment about flagging the scheduler in the config, we have a few hardcoded things we are moving to the config as well so I'll add it then
